src/CoreWCF.Primitives/src/CoreWCF/Security/SecuritySessionServerSettings.cs

```diff
@@ -1956,6 +1968,516 @@ public Task OpenAsync(CancellationToken token)
+                }
+
+                public T GetProperty<T>() where T : class
+                {
+                    T tObj = _serviceProvider.GetService<T>();
+                    if (tObj == null)
+                    {
+                        return IncomingChannel.GetProperty<T>();
+                    }
+                    else
+                    {
+                        return tObj;
+                    }
+                }
+
+                public Task OpenAsync()
+                {
+                    return Task.CompletedTask;
+                }
+
+                public override async Task OpenAsync(TimeSpan timeout)
+                {
+                    await base.OpenAsync(timeout);
+                    _channelDispatcher = await Settings.SecurityServiceDispatcher.
+                        GetInnerServiceChannelDispatcher(this);
+                }
+
+                public Task OpenAsync(CancellationToken token)
+                {
+                    return OpenAsync();
+                }
+
+                public Task SendAsync(Message message)
+                {
+                    return SendAsync(message, new TimeoutHelper(ServiceDefaults.SendTimeout).GetCancellationToken());
+                }
+
+                public Task SendAsync(Message message, CancellationToken token)
+                {
+                    //moved below code t dispatch async
+                     CheckOutputOpen();
+                    this.SecureApplicationMessage(ref message, null, token);
+                    // this.ChannelBinder.Send(message, timeoutHelper.RemainingTime());
+                    return IncomingChannel.SendAsync(message, token);
+                }
+            }
+            protected void CheckOutputOpen()
+            {
+                // ThrowIfClosedOrNotOpen();
+                lock (LocalLock)
+                {
+                    if (this.isOutputClosed)
+                    {
+                        throw DiagnosticUtility.ExceptionUtility.ThrowHelperWarning(new CommunicationException(SR.Format(SR.OutputNotExpected)));
+                    }
+                }
+            }
+
+            internal override void OnFaulted(Exception ex)
+            {
+                this.AbortCore();
+               // this.inputSessionCloseHandle.Fault(this);
+               // this.outputSessionCloseHandle.Fault(this);
+                base.OnFaulted(ex);
+            }
+        }
+
+
+        private class DuplexSessionRequestContext : RequestContextBase
+        {
+            private readonly IDuplexChannel _channel;
+            public DuplexSessionRequestContext(IDuplexChannel channel, Message request)
+                    : base(request, ServiceDefaults.CloseTimeout, ServiceDefaults.SendTimeout)
+            {
+                _channel = channel;
+            }
+            protected override void OnAbort()
+            {
+                
+            }
```

Comment:
When there's no method body, I prefer the style of placing the braces on the same line as method definition, e.g.
```
    protected override void OnAbort() { }
```